geomstats/stratified_geometry/spider.py

```diff
@@ -0,0 +1,303 @@
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...]
+            Boolean denoting if the SpiderPoint belongs to the Spider Set.
+        """
+        results = []
+        for single_point in point:
+            results += [
+                self._value_check(single_point)
+                and self._rays_check(single_point)
+                and self._zero_check(single_point)
+                and type(single_point) is SpiderPoint
+            ]
+        return gs.array(results)
+
+    def _rays_check(self, single_point):
+        r"""Check if a random point has the correct number of rays.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has a ray in the rays set.
+        """
+        if single_point.s not in list(range(self.rays + 1)):
+            return False
+        return True
+
+    @staticmethod
+    def _zero_check(single_point):
+        r"""Check if a random point satisfy the zero condition.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has zero length when it has zero ray.
+        """
+        if single_point.s == 0 and single_point.x != 0:
+            return False
+        return True
+
+    @staticmethod
+    def _value_check(single_point):
+        r"""Check if a random point has the correct length.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has a positive length when on non-zero ray.
+        """
+        if single_point.s != 0 and single_point.x <= 0:
+            return False
+        return True
+
+    @_vectorize_point((1, "point"))
+    def set_to_array(self, point):
+        r"""Turn a point into an array compatible with the dimension of the space.
+
+        Parameters
+        ----------
+        point : SpiderPoint or list of SpiderPoint, shape=[...]
+             Points to be checked.
+
+        Returns
+        -------
+        point_array : array-like, shape=[...,rays]
+            An array with the x parameter in the s position.
+        """
+        point_to_array = gs.zeros((len(point), self.rays))
+        for i, pt in enumerate(point):
+            point_to_array[i, pt.s - 1] = pt.x
+        return point_to_array
+
+
+class SpiderGeometry(PointSetGeometry):
+    """Geometry on the Spider, induced by the rays Geometry."""
+
+    def __init__(self, space, ambient_metric=EuclideanMetric(1)):
+        super(SpiderGeometry, self).__init__(space=space)
+        self.rays_geometry = ambient_metric
```

Comment:
It is confusing to have two different names: `ambient_metric` and `rays_geometry` that refer to the same object.

When a user initializes a new object and gives inputs (e.g. ambient_metric), they expect that these inputs are transformed into the object's attributes, without being renamed.

I would suggest:
`self.ray_metric = ray_metric`